errai-common-apt/src/main/java/org/jboss/errai/common/apt/ErraiAptExportedTypes.java

```diff
@@ -73,55 +72,74 @@ public static void init(final Types types,
     addLocalExportableTypesWhichHaveNotBeenExported();
   }
 
-  private static Map<String, Set<TypeMirror>> getExportedTypesFromExportFilesInExportFilesPackage() {
-    return exportFilesPackageElement(elements).map(ErraiAptExportedTypes::getExportedTypesFromExportFiles)
-            .orElseGet(HashMap::new);
+  private Map<String, Set<TypeMirror>> getExportedTypesFromExportFilesInExportFilesPackage() {
+    return exportFilesPackageElement(elements).map(this::getExportedTypesFromExportFiles).orElseGet(HashMap::new);
   }
 
-  private static Map<String, Set<TypeMirror>> getExportedTypesFromExportFiles(final PackageElement packageElement) {
+  private Map<String, Set<TypeMirror>> getExportedTypesFromExportFiles(final PackageElement packageElement) {
     return packageElement.getEnclosedElements()
             .stream()
-            .collect(groupingBy(ErraiAptExportedTypes::annotationName,
-                    flatMapping(ErraiAptExportedTypes::getExportedTypesFromExportFile, toSet())));
+            .collect(groupingBy(this::getAnnotationNameFromExportFileElement,
+                    flatMapping(this::getExportedTypesFromExportFileElement, toSet())));
+  }
+
+  private String getAnnotationNameFromExportFileElement(final Element e) {
+    return ExportFileName.decodeAnnotationClassNameFromExportFileName(e.asType().toString());
```

Comment:
So a fun thing I learned last week is that the `toString()` on `TypeElement` is not standard between compiler implementations. In Eclipse it will give you a string like "class org.acme.Foo" instead of just the name. The portable way to do this is to call `e.asType().getQualifiedName().toString()`.

Reply:
Oh, that's valuable info! Thanks, i'll totally change that :)